tests/dist/python/run_dist_objects.py

```diff
@@ -392,6 +393,248 @@ def dist_optimizer_check_store(g):
+                src_nodes_id = orig_nid_map[src_type][src_nodes_id]
+                dst_nodes_id = orig_nid_map[dst_type][dst_nodes_id]
+                eids = block.edges[etype].data[dgl.EID]
+                F.allclose(src_nodes_id, orig_u[eids])
+                F.allclose(dst_nodes_id, orig_v[eids])
+    del dataloader
+
+
+def distedgedataloader_test(g, batch_size, num_edges_to_sample, drop_last, shuffle,
+                            num_workers, orig_nid_map, orig_uv_map, num_negs):
+    if len(g.etypes) == 1:
+        train_eid = F.arange(0, num_edges_to_sample)
+    else:
+        train_eid = {g.etypes[-1]: F.arange(0, num_edges_to_sample)}
+
+    sampler = dgl.dataloading.MultiLayerNeighborSampler([5, 10])
+
+    dataloader = dgl.dataloading.DistEdgeDataLoader(
+         g,
+         train_eid,
+         sampler,
+         batch_size=batch_size,
+         negative_sampler=dgl.dataloading.negative_sampler.Uniform(num_negs) if num_negs > 0 else None,
+         shuffle=shuffle,,
+         drop_last=drop_last,
+         num_workers=num_workers,
+    )
+    for _ in range(2):
+        for _, sampled_data in zip(
+            range(0, num_edges_to_sample, batch_size), dataloader
+        ):
+            blocks = sampled_data[3 if num_negs > 0 else 2]
+            block = blocks[-1]
+            for src_type, etype, dst_type in block.canonical_etypes:
+                orig_u, orig_v = orig_uv_map[etype]
+                o_src, o_dst = block.edges(etype=etype)
+                src_nodes_id = block.srcnodes[src_type].data[dgl.NID][o_src]
+                dst_nodes_id = block.dstnodes[dst_type].data[dgl.NID][o_dst]
+                src_nodes_id = orig_nid_map[src_type][src_nodes_id]
+                dst_nodes_id = orig_nid_map[dst_type][dst_nodes_id]
+                eids = block.edges[etype].data[dgl.EID]
+                F.allclose(src_nodes_id, orig_u[eids])
+                F.allclose(dst_nodes_id, orig_v[eids])
+                if num_negs == 0:
+                    pos_pair_graph = sampled_data[1]
+                    assert np.all(
+                        F.asnumpy(block.dstnodes[dst_type].data[dgl.NID])
+                        == F.asnumpy(pos_pair_graph.nodes[dst_type].data[dgl.NID])
+                    )
+                else:
+                    pos_graph, neg_graph = sampled_data[1:3]
+                    assert np.all(
+                        F.asnumpy(block.dstnodes[dst_type].data[dgl.NID])
+                        == F.asnumpy(pos_graph.nodes[dst_type].data[dgl.NID])
+                    )
+                    assert np.all(
+                        F.asnumpy(block.dstnodes[dst_type].data[dgl.NID])
+                        == F.asnumpy(neg_graph.nodes[dst_type].data[dgl.NID])
+                    )
+                    assert pos_graph.num_edges() * num_negs == neg_graph.num_edges()
+    del dataloader
+
+def multi_distdataloader_test(g, dataloader_class):
+    num_items_to_sample=404
```

Comment:
could we remove magic numbers? how about use `node_split` to obtain `train_ids`?

Reply:
This an arbitrary number of relatively small size to make the test  runtime shorter.

Running it with much bigger training set will be make the test very long (the dist dataloader loaders runtime is already 10-15min with 2 nodes with those numbers)